aks/external-dns.tf

```diff
@@ -17,26 +25,25 @@ resource "helm_release" "external-dns" {
         name = "azure.secretName"
         value = "${kubernetes_secret.azure_dns_sp_creds.metadata.0.name}"
     }
-    
     set {
         name = "provider"
         value = "azure"
     }
 
-    set {
-        name = "azure.resourceGroup"
-        value = "${var.dns_zone_rg}"
-    }
-    
     set {
         name = "logLevel"
         value = "debug"
     }
 
+    set {
+        name = "azure.resourceGroup"
+        value = "${data.external.dns.result["resourceGroup"]}"
+    }
```

Comment:
Nice! I think having the resource group specified here again is an external-dns bug but no idea when they will fix it, if at all...